src/BenchmarkDotNet.Core/Portability/RuntimeInformation.cs

```diff
@@ -321,5 +321,35 @@ internal static ICollection<Antivirus> GetAntivirusProducts()
             return Array.Empty<Antivirus>();
 #endif
         }
+
+
+        internal static VirtualMachineHypervisor GetVirtualMachineHypervisor()
+        {
+            VirtualMachineHypervisor[] hypervisors = { new HyperV(), new VirtualBox(), new VMware() };
```

Comment:
It's a "best practice" in BenchmarkDotNet.Core mainly because we care about allocations. Internal BenchmarkDotNet allocations should never affect memory traffic in user code, so we are trying to avoid direct constructors if it's possible.